core/block.js

```diff
@@ -129,9 +128,24 @@ Blockly.Block = function(workspace, prototypeName, opt_id) {
    */
   this.collapsed_ = false;
 
-  /** @type {string|Blockly.Comment} */
```

Comment:
Even if this is deprecated, you want to keep the type as string|Blockly.Comment. 
Otherwise, ``this.comment`` is of type ``string`` in the superclass (Blockly.Block)
but ``this.comment`` is of type ``Blockly.Comment`` in the subclass (Blockly.BlockSvg)
And subclasses don't have the ability to just change the type of a property. They can only restrict.